src/courses/views.py

```diff
@@ -214,15 +218,70 @@ class CourseStudentCreate(NonPublicOnlyViewMixin, SuccessMessageMixin, LoginRequ
     success_url = reverse_lazy('quests:quests')
     success_message = "You have been added to the %(course)s course"
 
+    def get(self, request, *args, **kwargs):
+        # when accessing this view, check if we need a form at all, or can just create the studentcourse object using all defaults
+        # if simplified_course_registration is enabled in siteconfig and all three form fields are hidden, object should be created automatically
+        simpleregistration = SiteConfig.get().simplified_course_registration
+        form = self.get_form()
+
+        form_auto_create_condition = (
+            simpleregistration and
+            form.fields['semester'].widget.input_type == 'hidden' and
+            form.fields['block'].widget.input_type == 'hidden' and
+            form.fields['course'].widget.input_type == 'hidden'
+        )
+
+        if form_auto_create_condition:
+            # form_auto_create_condition == True means only one active/current semester, block, course object
+            # initialize semester, block, course objects in variables to create new CourseStudent with
+            auto_create_semester = SiteConfig.get().active_semester
+            auto_create_block = Block.objects.filter(active=True).first()
+            auto_create_course = Course.objects.filter(active=True).first()
+
+            # before object is created, check for existing identical CourseStudent object to avoid duplicate error
+            duplicate_coursestudent = CourseStudent.objects.filter(
```

Comment:
This might be simpler (more DRY) create the object within a try/except block, and catch the duplicate error, rather than repeating all this code here.

alternately, use [get_or_create()](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#get-or-create) method, and if created==False you know it already exists.

Reply:
get_or_create() method implemented!